### PR TITLE
Optimize parse patch read loop

### PR DESCRIPTION
**Pull request by techknowlogick**
Continuing #3159

### CODE REVIEW

File: models/git_diff_test.go [hunk 1]
```diff
@@ -139,3 +144,66 @@ func TestDiffLine_GetCommentSide(t *testing.T) {
...
+func benchParsePatch(b *testing.B, diffStr string) {
+	log.DelLogger("console")
+	log.DelLogger("file")
+	b.ResetTimer() //Disable logger for becnh
+	for i := 0; i < b.N; i++ {
+		ParsePatch(1000, 5000, 100, strings.NewReader(diffStr))
+	}
+}
+
+func BenchmarkParsePatchSimple(b *testing.B) {
```
Comment (techknowlogick) [inline models/git_diff_test.go:245]: @sapk you wrote these tests IIRC, would you be able to help take a look into why they are failing?

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/git_diff_test.go (hunk 1 above)
Comment (sapk) [reply models/git_diff_test.go:245]: It seems that ParsePatch doesn't work as it return no change (0 line, 0 file). Looking into it.

Comment (sapk) [reply models/git_diff_test.go:245]: I look at it a little more but it doesn't seems obvious. I will need to run it step by step.

Comment (sapk) [reply models/git_diff_test.go:245]: I hadn't time to look at it.

File: models/git_diff_test.go [hunk 2]
```diff
@@ -1,10 +1,15 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
 package models
```
[the remark this answers is not in the document]
Comment (zeripath) [reply models/git_diff_test.go:5]:
```suggestion

package models
```

**Comment by techknowlogick** [comment]
Closing this for now. If someone wants to re-open this please feel free.

**Pull request closed without merging**